src/topp/XTandemAdapter.cpp

```diff
@@ -427,6 +427,14 @@ class TOPPXTandemAdapter :
     search_parameters.missed_cleavages = getIntOption_("missed_cleavages");
     search_parameters.peak_mass_tolerance = getDoubleOption_("fragment_mass_tolerance");
     search_parameters.precursor_tolerance = getDoubleOption_("precursor_mass_tolerance");
+    if (getStringOption_("cleavage_site") == "[RK]|{P}")
+    {
+      search_parameters.enzyme = ProteinIdentification::TRYPSIN;
+    }
+    else if (getStringOption_("cleavage_site") == "[X]|[X]")
+    {
+      search_parameters.enzyme = ProteinIdentification::NO_ENZYME;
+    }
```

Comment:
I don't know if it's better to use the enzyme names or the specific cleavage sites, not just in XTandem but other cases. In this case, how about adding a cleavage_sites attribute into the SearchParameters, so that a new enzyme called "UNKNOWN_ENZYME" has the cleavage site information.